### ISSUE TITLE
Properly set compile_time on resources

### ISSUE THREAD

**Issue by tas50** [chef/cookstyle#673]
### What category of cop is this?:

Please try to select as few as possible.

- [ ] ChefDeprecations - A cop that alerts on a deprecation in the Chef Infra Client
- [x] ChefCorrectness - A cop that alerts when incorrect coding behavior is being used
- [ ] ChefSharing - A cop that alerts to missing functionality necessary for sharing cookbooks with others
- [ ] ChefStyle - A cop that alerts to a style best practice
- [ ] ChefModernize - A cop that alerts when a cookbook can be simplified or modernized with new functionality
- [ ] ChefEffortless - A cop that alerts on code that must be resolved to move to the Effortless pattern

### Describe the new cop:
You don't want to mix the compile_time property with .run_action on the resource. Just use compile_time and avoid the ruby mess.

### What it would trigger on:

```ruby
chef_gem 'deep_merge' do
  action :nothing
  compile_time true
end.run_action(:install)
```

### What it would autocorrect to if applicable

```ruby
chef_gem 'deep_merge' do
  action :install
  compile_time true
end
```

**Comment by lamont-granquist** [comment]
You might want to have another specific cop to fix someone with `action :nothing` and only `compile_time true` where the `action :nothing` can be deleted.

**Comment by tas50** [comment]
This is fixed in #687. It nukes the :nothing bit by copying in whatever action was in the run_action.
